generators/import-jdl/index.js

```diff
@@ -92,77 +92,159 @@ module.exports = class extends BaseGenerator {
-                            this.composeWith(require.resolve('../entity'), {
-                                force: this.options.force,
-                                debug: this.options.debug,
-                                regenerate: true,
-                                'skip-install': true,
-                                'skip-client': entity.definition.skipClient,
-                                'skip-server': entity.definition.skipServer,
-                                'no-fluent-methods': entity.definition.noFluentMethod,
-                                'skip-user-management': entity.definition.skipUserManagement,
-                                'skip-ui-grouping': this.options['skip-ui-grouping'],
-                                arguments: [entity.name],
-                            });
-                        }
-                    });
-                } catch (e) {
-                    this.debug('Error:', e);
-                    this.error(`Error while generating entities from parsed JDL\n${e}`);
+                    if (shouldGenerateApplications(this.jdlObject)) {
+                        this.log('Generating applications.');
+                        Object.keys(this.jdlObject.applications).forEach((application) => {
+                            this.log(`Generating application ${application.baseName}.`);
+                            generateApplicationFiles(this, this.jdlObject.applications[application]);
+                        });
+                    }
+                    this.log('Generating entities.');
+                    this.getExistingEntities()
+                        .filter(entity => this.changedEntities.includes(entity.name))
+                        .forEach((entity) => {
+                            if (shouldGenerateApplications(this.jdlObject) && entity.applications !== '*') {
+                                entity.applications.forEach((applicationName) => {
+                                    const applicationPath = path.join(
+                                        shelljs.pwd().stdout,
+                                        this.jdlObject.applications[applicationName].config.path || ''
+                                    );
+                                    generateEntityFilesInPath(this, entity, applicationPath);
+                                });
+                            } else {
+                                generateEntityFiles(this, entity);
+                            }
+                        });
+                } catch (error) {
+                    this.error(`While generating applications and entities from the parsed JDL\n${error}`);
                 }
             }
         };
     }
 
     end() {
-        if (!this.options['skip-install'] && !this.skipClient && !this.options['json-only']) {
+        if (!this.options['skip-install'] && !this.skipClient && !this.options['json-only']
+                && Object.keys(this.jdlObject.applications).length === 0) {
```

Comment:
this can be `!shouldGenerateApplications`

Reply:
Well spotted!